client/landing/gutenboarding/components/domain-picker/suggestion-item.tsx

```diff
@@ -28,13 +40,47 @@ const DomainPickerSuggestionItem: FunctionComponent< Props > = ( {
 	const domainName = domain.slice( 0, dotPos );
 	const domainTld = domain.slice( dotPos );
 
+	const { domainSearch } = useSelect( select => select( STORE_KEY ).getState() );
+	const fetchAlgo = `/domains/search/${ DOMAIN_SUGGESTION_VENDOR }/${ FLOW_ID }`;
+	const [ previousDomain, setPreviousDomain ] = useState< string | undefined >();
+	const [ previousRailcarId, setPreviousRailcarId ] = useState< string | undefined >();
+
+	useEffect( () => {
+		// Only record TrainTracks render event when the domain name changes
+		if ( domain !== previousDomain || previousRailcarId !== railcarId ) {
+			recordTrainTracksRender( {
+				fetchAlgo,
+				query: domainSearch,
+				railcarId,
+				result: isRecommended ? domain + '#recommended' : domain,
+				uiPosition,
```

Comment:
This is the domain's position on the list — to me, "ui position" sounds a little bit like the position of the whole suggestion list.

We'll have the same list appear in few places, in immediate future we'll have the domain dropdown (what we have now in `master`) and "more domains modal" (https://github.com/Automattic/wp-calypso/issues/41081). Possibly some additional domain steps and other places as well. We should likely include that info in tracking event?

Reply:
Absolutely. I think if we add another param listed in PCYsg-bor-p2 `ui_algo` we can use that to denote the unique place where the list appears. E.g. we could have:

* `/gutenboarding/domain-popover`
* `/gutenboarding/domain-expanded`
* `/gutenboarding/domain-step`

Etc.